source/fundamentals/class-mapping.txt

```diff
@@ -0,0 +1,326 @@
+level by using attributes with the class, or by calling methods while registering
+a class map.
+
+Ignore Extra Elements
+~~~~~~~~~~~~~~~~~~~~~
+
+When a BSON document is deserialized to a {+language+} class, the {+driver-short+}
+looks at the name of each field in the document and tries to find a matching property
+name in the class. By default, if a field in the document doesn't have a match in the class,
+the driver throws an exception. However, you can choose to ignore any elements that do 
+not have a matching class property by using the ``BsonIgnoreExtraElements`` attribute.
+This prevents the driver from throwing an exception, and maps any other fields
+that have matching class properties.
+
+The following example shows how to add a ``BsonIgnoreExtraElements`` attribute
+to a class.
+
+.. code-block:: csharp
+
+   [BsonIgnoreExtraElements]
+   public class Person
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+   }
+
+You can also ignore any extra elements when registering a class map:
+
+.. code-block:: csharp
+   
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+        classmap.AutoMap();
+        classmap.SetIgnoreExtraElements(true);
+   });
+
+Using Class Discriminators
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can specify **discriminators** to help identify **polymorphic** classes that
+are serialized to the same collection. Polymorphic classes are classes that
+inherit properties and methods from a parent class. A discriminator is an
+element that's added to a document to identify which class the document was
+serialized from.
+
+You can specify a discriminator using the ``BsonDiscriminator`` attribute as
+follows:
+
+.. code-block:: csharp
+
+   [BsonDiscriminator("personClass")]
+   public class Person 
+   {
+       public string Name { get; set; }
+       public int Age { get; set; }
+       public List<string> Hobbies {get; set;}
+   }
+
+You can also specify a discriminator when registering a class map as follows:
+
+.. code-block:: csharp
+
+   BsonClassMap.RegisterClassMap<Person>(classmap => 
+   {
+       classmap.AutoMap();
+       classmap.SetDiscriminator("personClass");
+   });
+
+.. note:: 
+
+   The driver stores your discriminator in BSON with the field name ``_t``.
+
+A document from the ``Person`` class with the "personClass" discriminator is
+stored in the collection as follows:
+
+.. code-block:: json
+
+   { "_id": "...", "_t": "personClass", "name": "...", "age": "...", "hobbies": [...]}
+
+.. TODO: Link to page on polymorphism/discriminators
+
+Mapping with Constructors
+-------------------------
+
+By default the {+driver-short+} requires a class to have a constructor with no 
+arguments in order to automatically create a class map. However, you can
+instruct the driver to automatically map the class using a constructor that takes one
+or more arguments.
```

Comment:
How does this change the serialization behavior/how does it change the way documents appear in BSON?